docs/arch/controller.md

```diff
@@ -0,0 +1,40 @@
+# Architecture Document for Controller Component
+
+```mermaid
+sequenceDiagram
+    actor User
+    User->>Controller: sends pipeline jobs as YAML and repository link
+    alt is request malformed
+        Controller-->>User: Nok
+    else is well
+        Controller-->>User: Ok
+    end
+    Controller->>Database: saves pipeline in database
+    loop over steps
+        Controller-->>Scheduler: sends step (over GRPC)
+        Scheduler-->>Controller: job succeeded or not
+    end
+    User->>Controller: get pipeline state
+```
+
+## Glossary
+- action : list of steps, a declarative file can only contain one action.
+- step : a step is the most atomic component of our pipeline.
+
+## Component Description
+- Controller component is an app directly interacting with the Monitor and the Scheduler.
+It can receive 1 Action at a time through a REST API exposed directly on Internet. 
+
+## Goals of the Controller
+- Users can use the API to control/manage Actions/Steps
```

Comment:
Why say "users" when it can be a user, a bot/cron/... or the monitor(?) or whatever?

Reply:
U can use a passive form like "a controller manages steps through API calls"

Reply:
I'm not sure it's a "goal" ("what" it does) I think it's more a "how" it does it?

Reply:
User is to be understood as actor. It's not our business if it's a bot or a user interacting with the controller through its API. We are going to change user to actor if you prefer